addons/project/report/project_task_burndown_chart_report.py

```diff
@@ -96,44 +96,70 @@ def _search(self, domain, offset=0, limit=None, order=None, **kwargs):
                         stage_id,
                         is_closed
                    FROM (
-                            -- Gathers the stage_ids history per task_id. This query gets:
-                            -- * All changes except the last one for those for which we have at least a mail
-                            --   message and a mail tracking value on project.task stage_id.
-                            -- * The stage at creation for those for which we do not have any mail message and a
-                            --   mail tracking value on project.task stage_id.
                             SELECT DISTINCT task_id,
                                    allocated_hours,
                                    project_id,
                                    %(date_begin)s as date_begin,
                                    %(date_end)s as date_end,
-                                   first_value(stage_id) OVER task_date_begin_window AS stage_id,
+                                   stage_id,
                                    is_closed
                               FROM (
+                                     -- Gathers the stage_ids history per task_id from the stage history tracking field.
```

Comment:
Don't use stage_history tracking. I feel like we should use a `CROSS JOIN LATERAL jsonb_each`